### PR TITLE
Support era to resolve date string

### LINKED ISSUE #164: problem of date format in ja-JP
**Code:**
```
  const IntlPolyfill = require('intl');
  DateTimeFormat = IntlPolyfill.DateTimeFormat;
  require('intl/locale-data/jsonp/ja');

console.log(
  new DateTimeFormat('ja-JP-u-ca-japanese', {
    day: 'numeric',
    month: 'long',
    year: 'numeric',
    era: 'short'
}).format(Date.now()));
```

**Expect:**
平成28年4月21日
**But:**
12016年4月月21日

**Version:**
1.1.0

**PS.**
this works fine
`DateTimeFormat = global.Intl.DateTimeFormat;`

### PR DESCRIPTION
**Pull request by watilde**
Fixes #164

### CODE REVIEW

File: src/12.datetimeformat.js [hunk 1]
```diff
@@ -854,7 +854,14 @@ function CreateDateTimeParts(dateTimeFormat, x) {
                     fv = ''; // TODO
                     break;
 
-                    // TODO: Era
+                  case 'era':
+                    try {
+                      fv = resolveDateString(localeData, ca, 'eras', f, tm['[['+ p +']]']);
```
Comment (caridy) [inline src/12.datetimeformat.js]: interesting, the spec doesn't list `"era"` in the list of things to keep in mind: https://tc39.github.io/ecma402/#sec-datetimeformat-abstracts

I wonder if we need to revisit the spec in that regards.

Comment (caridy) [inline src/12.datetimeformat.js]: When I try this in chrome, this is what I'm getting:

```
new Intl.DateTimeFormat('ja-JP-u-ca-japanese', {
    day: 'numeric',
    month: 'long',
    year: 'numeric',
    era: 'short'
}).resolvedOptions();
// output:
Object {
  calendar: "japanese"
  day: "numeric"
  era: "short"
  locale: "ja-u-ca-japanese"
  month: "numeric"
  numberingSystem: "latn"
  timeZone: "America/New_York"
  year: "numeric"
}
```

Comment (watilde) [inline src/12.datetimeformat.js]: Hmm. I tried same code with the above and they seems have something defference between `Intl.js` and `chrome`: https://gist.github.com/watilde/d0406a357ed8367d3cbd12f32dfdcf1e
I will try to take a look into some related things with this more.

**Comment by watilde** [comment]
Rebased and then passed all tests. We still need this patch to fix the issue, but the `era` part is not correct yet. I will continue to fix it.

Current results:
+ master branch: `12016年4月30日`
+ this patch: `白雉2016年4月30日`
+ should be: `平成2016年4月30日`

**Comment by watilde** [comment]
Ok so this patch solve an issue related #164, without the cases which are using the [BCP 47 Extension](http://www.unicode.org/reports/tr35/#Locale_Extension_Key_and_Type_Data) data like `ja-JP-u-ca-japanese` because we don't have the extension cldr data. Hence, this result is correct:

```js
var DateTimeFormat = IntlPolyfill.DateTimeFormat;

console.log(
  new Intl.DateTimeFormat('ja-JP', {
    year: 'numeric',
    era: 'short'
  }).format(Date.now())
);
// => 西暦2016年
```

Will write a feed back to `12.1Abstract Operations For DateTimeFormat Objects` as well if the thing is missing in the doc. Thanks!

**Comment by caridy** [comment]
thanks @watilde, this is now merged.

**Comment by caridy** [comment]
@watilde lets work to add that to the spec text, let's start by creating a new issue in https://github.com/tc39/ecma402/issues/new, and I will follow up with others.

**Comment by watilde** [comment]
@caridy Thanks Caridy! I've already created an issue on [here](https://github.com/tc39/ecma402/issues/88), then now I will try to add it to the spec 😺

**Pull request merged**